jax/_src/lax/slicing.py

```diff
@@ -64,7 +62,7 @@ def slice(operand: Array, start_indices: Sequence[int],
                       limit_indices=tuple(limit_indices),
                       strides=None if strides is None else tuple(strides))
 
-def dynamic_slice(operand: Array, start_indices: Sequence[Array],
+def dynamic_slice(operand: Array, start_indices: Sequence[ArrayLike],
```

Comment:
Should this also be `start_indices: Union[Array, Sequence[ArrayLike]]`?

Reply:
`ArrayLike` includes scalars which don't have `__len__`, so it would fail to typecheck. We could use `Union[Array, Sequence[ArrayLike]]`, what do you think?

Reply:
SGTM

Reply:
Done